Classes/Console/Command/DatabaseCommandController.php

```diff
@@ -163,15 +164,30 @@ public function importCommand($interactive = false)
      * If this imposes a security risk for you, then refrain from using this command!
      *
      * @param array $excludeTables Comma-separated list of table names to exclude from the export
+     * @param bool $excludeVolatile Exclude cache and session tables
      */
-    public function exportCommand(array $excludeTables = [])
+    public function exportCommand(array $excludeTables = [], $excludeVolatile = false)
     {
         $dbConfig = $this->connectionConfiguration->build();
         $additionalArguments = [
             '--opt',
             '--single-transaction',
         ];
 
+        if ($this->output->getSymfonyConsoleOutput()->isVerbose()) {
+            $additionalArguments[] = '--verbose';
+        }
+
+        if ($excludeVolatile) {
+            array_push($excludeTables, 'fe_sessions', 'be_sessions');
+            array_push($excludeTables, 'cache_md5params', 'cache_treelist');
+            foreach ($GLOBALS['TYPO3_CONF_VARS']['SYS']['caching']['cacheConfigurations'] as $name => $configuration) {
+                if (is_a($configuration['backend'], Typo3DatabaseBackend::class, true)) {
+                    array_push($excludeTables, 'cf_' . $name, 'cf_' . $name . '_tags');
```

Comment:
This won't work for custom table names which can be configured per cache. You'd actually need an instance of the backend and use its getters to retrieve the table names but I assume that's way too much overhead here.

Reply:
That is why I exactly don't want to maintain such a list in command code.

Table names change over time, might be reconfigured and thus increases maintenance and has decreased usefulness.

Also, even when not taking into account future changes, this is still technically wrong, when taking into account that fact that multiple database connections can be configured and every table might be on an individual connection.

If we can't agree on having a file format for that, I'd rather implement a generic wildcard feature, which would make `--exclude-tables` more flexible to use.

And even with that, we are still lacking a way to deal with multiple connections. Considering both I don't see an easier solution than having a configuration file to define what should be exported/imported from where exactly

Reply:
good point. I look into it. The dump thing is that I can't simply ask the CacheManager for all instances since that is all protected.

Reply:
@helhum the thing is: this solution works in a most if not all instances. (~ not sure how many instances actually use multiple connections and also change cache table names). Yes the 4 hardcoded tables aren't pretty... i though about just excluding the cache tables since that actually all that is really important for a quick dump. Also, the dump command does not break if a table that is excluded does not exist so there is really no risk at all.

Have you looked into my slimdump experiment? If you really want I can extend the command with that ~ although it will be diffecult without breaking backward compatibility. I'm not sure if that will reduce maintainance but creating an own solution is probably a lot more maintainance than including an external project.

Reply:
> Have you looked into my slimdump experiment?

slimdump experiment?

Reply:
@mbrodala Just checked. I can't find a way to configure cache table names. as far as I can see they are always inferred from cache name.

@Nemo64 therefore I don't see much benefit in implementing `--exclude-cache`, when one could also write `--exclude-tables 'cf_*,cache_*'` given wildcards are implemented of course.

So I'd rather implement wildcards as first step before considering additional options.

Fine?

Reply:
You are right, I must have mixed this up with a very old Caching Framework state where one was required to set table names.

Reply:
@helhum wildcards in exclude-tables are an idea I can live with.